acquisitions/src/main/resources/thunderjet/mod-gobi/features/gobi-api-tests.feature

```diff
@@ -80,3 +80,214 @@ Feature: mod-gobi api tests
     And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
     When method DELETE
     Then status 200
+
+  Scenario: Try to send invalid requests and check responses
+    * def valid_mapping = read('classpath:samples/mod-gobi/unlisted-print-serial.json')
+    * def invalid_mapping_type = read('classpath:samples/mod-gobi/unlisted-print-serial-custom.json')
+    * def invalid_mapping_field = read('classpath:samples/mod-gobi/unlisted-print-serial-2.json')
+    * def invalid_mapping_translation = read('classpath:samples/mod-gobi/unlisted-print-serial-1.json')
+
+    # add mapping configuration with invalid order type
+    Given path '/gobi/orders/custom-mappings'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    And request invalid_mapping_type
+    When method POST
+    Then status 400
+
+    # verify the new mapping changes were not applied
+    Given path '/gobi/orders/custom-mappings/UnlistedPrintSerial'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    When method DELETE
+    Then status 404
+
+    # add mapping configuration with invalid field name
+    Given path '/gobi/orders/custom-mappings'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    And request invalid_mapping_field
+    When method POST
+    Then status 400
+
+    # verify the new mapping changes were not applied
+    Given path '/gobi/orders/custom-mappings/UnlistedPrintSerial'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    When method DELETE
+    Then status 404
+
+    # add mapping configuration with invalid translation
+    Given path '/gobi/orders/custom-mappings'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    And request invalid_mapping_translation
+    When method POST
+    Then status 400
+
+    # verify the new mapping changes were not applied
+    Given path '/gobi/orders/custom-mappings/UnlistedPrintSerial'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    When method DELETE
+    Then status 404
+
+  Scenario: Make sure API methods behave correctly in different situations
+    * def valid_mapping = read('classpath:samples/mod-gobi/unlisted-print-serial.json')
+
+    # Verify original mapping
+    Given path '/gobi/orders/custom-mappings/UnlistedPrintSerial'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    When method GET
+    Then status 200
+    And match response.orderMappings.orderType == 'UnlistedPrintSerial'
+    And match response.orderMappings.mappings[1].dataSource.default == "true"
+    And match response.orderMappings.mappings[2].dataSource.default == "false"
```

Comment:
not relates to this PR, but seems found issue in mod-gobi.
https://github.com/folio-org/mod-gobi/blob/master/src/main/java/org/folio/rest/service/GobiCustomMappingsService.java#L38
This method always loads default mappings and ignore custom mappings, we can fix it in scope of:
https://folio-org.atlassian.net/browse/MODGOBI-203